File: LeetCode/C#/9. Palindrome Number.cs
public class Solution {
    public bool IsPalindrome(int x) {
        var num1 = x;
        var num2 = 0;
        
        while(x > 0)
        {
            num2 = num2 * 10 + x % 10;
            x /= 10;
        }
        
        return num1 == num2;  
    }
}

File: LeetCode/Java/9. Palindrome Number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }
        
        int originalNum = x;
        int reversedNum = 0;

        while(x != 0) {
            int lastDigit = x % 10;
            reversedNum = (reversedNum * 10) + lastDigit;
            x /= 10;
        }

        return originalNum == reversedNum;
    }
}